solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java

```diff
@@ -741,16 +766,29 @@ public boolean split(ClusterState clusterState, ZkNodeProps message, NamedList<O
         } else {
           ccc.offerStateUpdate(Utils.toJSON(m));
         }
+
```

Comment:
This fixes a race condition between setting the shards in RECOVERY state and adding more replicas. Thanks @andyvuong for this fix.

Reply:
I feel that race deserves its own issue/pr.  @andyvuong if you don't want to create it, we can.

Reply:
@dsmiley @bruno-roustant sure - i can create one, thanks.